src/components/main-view/main-view.jsx

```diff
@@ -28,56 +38,106 @@ export const MainView = () => {
       .then((response) => response.json())
       .then((movies) => {
         setMovies(movies);
-
-        console.log(movies);
       });
   }, [token]);
 
   return (
-    <Row>
-      <Button
-        onClick={() => {
+    <BrowserRouter>
+      <NavigationBar
+        user={user}
+        onLoggedOut={() => {
           setUser(null);
           setToken(null);
           localStorage.clear();
+          // redirecting to login page
+          window.location.href = "/login";
         }}
-        variant="link"
-      >
-        Logout
-      </Button>
-      {!user ? (
-        <Col md={5}>
-          <LoginView
-            onLoggedIn={(user, token) => {
-              setUser(user);
-              setToken(token);
-            }}
-          />
-          or
-          <SignupView />
-        </Col>
-      ) : selectedMovie ? (
-        <Col md={8} style={{ border: "1px solid black" }}>
-          <MovieView
-            movie={selectedMovie}
-            onBackClick={() => setSelectedMovie(null)}
-          />
-        </Col>
-      ) : movies.length === 0 ? (
-        <div>The list is empty!</div>
-      ) : (
-        movies.map((movie) => (
-          <Col className="mb-5 col-12 col-md-6 col-lg-4" key={movie._id}>
-            <MovieCard
-              key={movie._id}
-              movie={movie}
-              onMovieClick={(newSelectedMovie) => {
-                setSelectedMovie(newSelectedMovie);
-              }}
-            />
-          </Col>
-        ))
-      )}
-    </Row>
+      />
+      <Row>
+        <Routes>
+          <Route
+            path="/login"
+            element={
+              <>
+                {!user ? (
+                  <Col md={5}>
+                    <LoginView
+                      onLoggedIn={(user, token) => {
+                        setUser(user);
+                        setToken(token);
+                      }}
+                    />
+                  </Col>
+                ) : (
+                  <Navigate to="/" />
+                )}
+              </>
+            }
+          ></Route>
+          <Route
+            path="/signup"
+            element={
+              <>
+                <SignupView />
+              </>
+            }
+          ></Route>
+          <Route
+            path="/"
+            element={
+              <>
+                {!user ? (
+                  <Navigate to="/login" replace />
+                ) : (
+                  <>
+                    {movies.map((movie) => (
+                      <Col
+                        className="mb-5 col-12 col-md-6 col-lg-4"
+                        key={movie._id}
+                        movie={movie}
+                      >
+                        <MovieCard
+                          key={movie._id}
+                          movie={movie}
+                          onToggleFavorite={handleToggleFavorite}
```

Comment:
the prop to add here
`isFavorite={user.FavoriteMovies.includes(movie._id)}`

Reply:
done